expertsystem/data.py

```diff
@@ -63,11 +63,20 @@ class Spin(abc.Hashable):
     def __init__(self, magnitude: float, projection: float) -> None:
         magnitude = float(magnitude)
         projection = float(projection)
+        if magnitude % 0.5 != 0.0:
```

Comment:
Also add a check for `magnitude >= 0.0`

Reply:
Not needed because of exception upon `abs(projection) > magnitude`

Reply:
True, tricky :smile: